packages/design-system/src/components/contextMenu/menu.js

```diff
@@ -39,19 +39,33 @@ import { useContextMenu } from './contextMenuProvider';
 
 const MenuWrapper = styled.div(
   ({ theme }) => css`
-    padding: ${({ $isIconMenu }) => ($isIconMenu ? '4px 3px' : '8px 0')};
-    background-color: ${theme.colors.bg.primary};
-    border-radius: ${theme.borders.radius.small};
+    background-color: ${({ $isSecondary }) =>
+      $isSecondary ? theme.colors.bg.secondary : theme.colors.bg.primary};
+    border-radius: ${theme.borders.radius.medium};
     border: 1px solid ${theme.colors.border.disable};
-    width: ${({ $isIconMenu }) => ($isIconMenu ? 40 : 210)}px;
+    gap: 6px;
+    display: flex;
```

Comment:
This is where I moved to a flexbox layout with gaps rather than margins. That makes it a lot easier for switching directions (and works out of the box for RTL).